test/support/publish_dsl.rb

```diff
@@ -25,11 +25,191 @@ def publish_wait
       page.wont_have_css 'publish-spinner'
     end
 
+    def publish_wait_for_title
+      find('#title')
+    end
+
+    def publish_wait_for_episodes_link
+      find('a', text: '0 Episodes in Series')
+    end
+
+    def publish_wait_for_success
+      find('div.success')
+    end
+
     def publish_visit(url)
       publish_login!
       visit url
       publish_wait
     end
 
+    def delete_test_series_and_episodes!
+      publish_login!
+      series_list_url = CONFIG.PUBLISH_HOST + '/search;tab=series;perPage=50'
+      visit series_list_url
+      begin
+        find('button', text: /^View All \d+/, wait: 30) # wait for page to laod
+      rescue
+        puts "Less than 50 series to clean up" if debug?
+      end
+      series_links = {}
+      page.all('a', text: /^Test Series \w{20}$/).each do |link|
+        series_url = link[:href]
+        next if series_links[series_url]
+        series_links[series_url] = true
+      end
+      series_links.keys.each do |series_url|
+        puts "Cleaning up #{series_url}" if debug?
+        delete_all_episodes_in_series!(series_url)
+        puts "All episodes deleted" if debug?
+        visit series_url
+        click_button('Delete')
+        click_button('Okay')
+      end
+      visit series_list_url
+    end
+
+    def delete_all_episodes_in_series!(series_url)
+      episode_list_url = series_url + '/list'
+      visit episode_list_url
+      begin
+        find('a', text: /Search among these episodes/)
+      rescue
+        return # no episodes
+      end
+      episode_links = {}
+      page.all('a', text: /^Test Episode \w{20}$/).each do |link|
+        episode_url = link[:href]
+        next if episode_links[episode_url]
+        episode_links[episode_url] = true
+      end
+      episode_links.keys.each do |episode_url|
+        puts "Deleting #{episode_url}" if debug?
+        m = episode_url.match(/\/story\/(\d+)$/)
+        ep_id = m[1]
+        # there's a wonky bug with Poltergeist and Angular where the modal Okay
+        # button does not trigger a DELETE call. So we do it ourselves.
```

Comment:
I'm a little bothered that it is working for the series delete modal okay button but not the story.

Reply:
Agreed.

Reply:
I also agree. I found it strange that it worked in one place and not another. For awhile I thought it was a config issue with Poltergeist and/or phantomjs, but there's a JS error that gets generated on the page. I will open a separate ticket with that error as a tickler to fix the underlying JS issue so we can get rid of this workaround in the tests.

Reply:
https://github.com/PRX/publish.prx.org/issues/479